scripts/pylib/twister/twisterlib/testsuite.py

```diff
@@ -391,10 +391,14 @@ def __init__(self, suite_root, suite_path, name, data=None):
         """
 
         workdir = os.path.relpath(suite_path, suite_root)
-        self.name = self.get_unique(suite_root, workdir, name)
+
+        assert self.check_suite_name(name, suite_root, workdir)
```

Comment:
Assert should only be used for testing and debugging — not in production environments.
Here the assert can be removed, because in `check_suite_name` an exception is raised.